workspaces/src/worker/impls.rs

```diff
@@ -116,7 +117,15 @@ where
 
     /// View the latest block from the network
     pub async fn view_latest_block(&self) -> anyhow::Result<Block> {
-        self.client().view_block(None).await.map(Into::into)
+        self.view_block(BlockReference::latest()).await
+    }
+
+    /// View the block from the network given the block reference to it.
+    pub async fn view_block(&self, block_ref: BlockReference) -> anyhow::Result<Block> {
```

Comment:
This seems unergonomic to use, wonder if it's a better solution to split this function into this and `view_block_at_height`, `view_block_at_hash` or something so that someone doesn't have to write:

```rust
BlockReference::BlockId(BlockId::Height(_))
```

as a parameter and expose all of these enum types?

Reply:
they can directly do:
```
view_block(BlockId::Height(...).into())
```
to avoid importing `BlockReference` but that might be a little clunky. But BlockId should be the most frequently used case, so adding `view_block_at_height` and `view_block_at_hash` is not such a bad idea

Reply:
Yeah, I'm not sure what the best pattern would be for these, but maybe we should come up with a plan and stay consistent throughout the repo since this pattern will be common

Reply:
I think what we have right now is pretty consistent where we have `BlockReference` being the general input, where we have helper methods for height and hash for the more used paths. I just gotta add the general method to importing contracts to make it consistent, but I'll push that to a separate PR since it's a big change in of itself